### PR TITLE
Typo3 12 compatibility

### PR DESCRIPTION
**Pull request by hojalatheef**
- Updated php cs fixer configurations
- TYPO3 11 and 12 pipeline actions updated

### CODE REVIEW

File: .github/workflows/typo3_v11ci.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Adapted from https://github.com/TYPO3GmbH/blog/blob/master/.github/workflows/ci.yml
+name: CI
+
+on: [push, pull_request]
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v11ci.yml:4]: pull_request should be enough

File: .github/workflows/typo3_v11ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  build:
+
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v11ci.yml:8]: Remove return. If you want you can seperate properties on same level with a return like after runs-on

File: .github/workflows/typo3_v11ci.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        typo3: [^11.5]
+        php: ['7.4', '8.0']
+
+    steps:
+
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v11ci.yml:16]: remove return

File: .github/workflows/typo3_v11ci.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup PHP ${{ matrix.php }}, with composer and extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: mbstring, dom, zip, mysql
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v11ci.yml:24]: I can only find linting and php-cs-fixer. No func. test. So mysql is not needed here

File: .github/workflows/typo3_v12ci.yml [hunk 5]
```diff
@@ -9,18 +9,19 @@ jobs:
...
-        typo3: [^9.5, ^10.4]
-        php: ['7.2', '7.3', '7.4']
+        typo3: [^12.4]
+        php: [8.1']
 
     steps:
 
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v12ci.yml:18]: perfect

File: .github/workflows/typo3_v12ci.yml [hunk 6]
```diff
@@ -9,18 +9,19 @@ jobs:
...
 
-      - name: Set up PHP Version ${{ matrix.php }}
-        run: |
-          sudo update-alternatives --set php /usr/bin/php${{ matrix.php }}
-          php -v
+      - name: Setup PHP ${{ matrix.php }}, with composer and extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: mbstring, dom, zip, mysql
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v12ci.yml:24]: PLease add mysql only if needed

File: .php-cs-fixer.cache [hunk 7]
```diff
@@ -0,0 +1 @@
...dernize_types_casting":true,"native_function_casing":true,"new_with_braces":true,"no_alias_functions":true,"no_blank_lines_after_phpdoc":true,"no_empty_phpdoc":true,"no_empty_statement":true,"no_extra_blank_lines":true,"no_leading_import_slash":true,"no_leading_namespace_whitespace":true,"no_null_property_initialization":true,"no_short_bool_cast":true,"no_singleline_whitespace_before_semicolons":true,"no_superfluous_elseif":true,"no_trailing_comma_in_singleline_array":true,"no_unneeded_control_parentheses":true,"no_unused_imports":true,"no_useless_else":true,"no_whitespace_in_blank_line":true,"ordered_imports":true,"php_unit_construct":true,"php_unit_mock_short_will_return":true,"php_unit_test_case_static_method_calls":{"call_type":"self"},"phpdoc_no_access":true,"phpdoc_no_empty_return":true,"phpdoc_no_package":true,"phpdoc_scalar":true,"phpdoc_trim":true,"phpdoc_types":true,"phpdoc_types_order":{"null_adjustment":"always_last","sort_algorithm":"none"},"return_type_declaration":{"space_before":"none"},"single_quote":true,"single_trait_insert_per_statement":true,"whitespace_after_comma_in_array":true},"hashes":{"Classes\/Domain\/Model\/News.php":"ce99930a3116c384b151c66448c83189"}}
```
Comment (sfroemkenjw) [inline .php-cs-fixer.cache:1]: Please add this file into .gitignore

File: composer.json [hunk 8]
```diff
@@ -24,12 +24,11 @@
 		"source": "https://github.com/jweiland-net/jwnewsevent"
 	},
 	"require": {
-		"typo3/cms-core": "^9.5.17 || ^10.4.2",
-		"georgringer/news": "^8.0"
+		"typo3/cms-core": "^11.5.33 || ^12.4.8",
```
Comment (sfroemkenjw) [inline composer.json:27]: Nice, you have updated the deps to current secure TYPO3 versions.

**Review by sfroemkenjw — CHANGES_REQUESTED** [review]

File: .github/workflows/typo3_v11ci.yml (hunk 1 above)
Comment (hojalatheef) [reply .github/workflows/typo3_v11ci.yml:4]: Done

File: .github/workflows/typo3_v11ci.yml (hunk 2 above)
Comment (hojalatheef) [reply .github/workflows/typo3_v11ci.yml]: Done

File: .github/workflows/typo3_v11ci.yml (hunk 3 above)
Comment (hojalatheef) [reply .github/workflows/typo3_v11ci.yml]: Done

File: .github/workflows/typo3_v11ci.yml (hunk 4 above)
Comment (hojalatheef) [reply .github/workflows/typo3_v11ci.yml]: Done

File: .github/workflows/typo3_v12ci.yml (hunk 6 above)
Comment (hojalatheef) [reply .github/workflows/typo3_v12ci.yml:24]: Done

File: .php-cs-fixer.cache (hunk 7 above)
Comment (hojalatheef) [reply .php-cs-fixer.cache]: Done

File: .github/workflows/typo3_v11ci.yml [hunk 9]
```diff
@@ -0,0 +1,36 @@
+# Adapted from https://github.com/TYPO3GmbH/blog/blob/master/.github/workflows/ci.yml
+name: CI
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v11ci.yml:2]: Can you please add propper naming to differ between TYPO3 11 and 12 testing?

File: .github/workflows/typo3_v12ci.yml [hunk 10]
```diff
@@ -1,26 +1,27 @@
 # Adapted from https://github.com/TYPO3GmbH/blog/blob/master/.github/workflows/ci.yml
 name: CI
```
Comment (sfroemkenjw) [inline .github/workflows/typo3_v12ci.yml:2]: Can you please add propper naming to differ between TYPO3 11 and 12 testing?

**Review by sfroemkenjw — CHANGES_REQUESTED** [review]

File: .github/workflows/typo3_v11ci.yml (hunk 9 above)
Comment (hojalatheef) [reply .github/workflows/typo3_v11ci.yml]: Done

**Pull request merged**